client.py

```diff
@@ -1232,20 +1236,54 @@
 		"""
 			Punish the user for swearing.
 		"""
-		if any(swear in ewutils.flattenTokenListToString(content_tolower) for swear in ewcfg.curse_words):
+		if (any(swear in content_tolower_string for swear in ewcfg.curse_words.keys())):
+			
+			swear_multiplier = 0
+			
+			#print(content_tolower_string)
+	
 			playermodel = ewplayer.EwPlayer(id_user=message.author.id)
 			if message.server != None:
 				usermodel = EwUser(id_user=message.author.id, id_server=playermodel.id_server)
 			else:
 				usermodel = None
-			
+
+			market_data = EwMarket(id_server=message.server.id)
+
+			# gather all the swear words the user typed.
+			for swear in ewcfg.curse_words.keys():
+				
+				if swear == "buster" and usermodel.faction == ewcfg.faction_rowdys:
+					continue
+				if swear == "kraker" and usermodel.faction == ewcfg.faction_killers:
+					continue
+					
+				swear_count = content_tolower_string.count(swear)
+				
+				# A niche scenario. If the user types either of these emotes at least once, then 'fuck' will not be detected.
+				if swear == "fuck" and (content_tolower_string.count('<rowdyfucker431275088076079105>') > 0 or content_tolower_string.count('<fucker431424220837183489>') > 0):
+					#print('emote skipped over')
+					continue
+				
+				for i in range(swear_count):
+					swear_multiplier += ewcfg.curse_words[swear]
+					
+					market_data.global_swear_jar += 1
+
+					usermodel.swear_jar += 1
+
+			market_data.persist()
+
 			if usermodel != None:
-				usermodel.swear_jar += 1
-				usermodel.slimecoin = max(0, usermodel.slimecoin - 1000000000) # 1 billion slimecoin fine for swearing
+					
+				usermodel.slimecoin = max(0, (usermodel.slimecoin - (usermodel.swear_jar * swear_multiplier * 10000))) # fine the user for swearing, 1 billion slimecoin for every swear in their swear jar
```

Comment:
we should use change_slimecoin() for this and track how much each player has actually been fined over their lifetime